### Задание 5.1
Когда используется критерий Стьюдента, а когда Z –критерий?

### Ответ
Z-тест используется, на основе нормального распределения и при известной дисперсии.

### Задание 5.2
Проведите тест гипотезы. Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм.
Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр
оказался равным 17.5 мм, а дисперсия известна и равна 4 кв. мм.

### Решение

Гипотеза $H_0$: $\mu = 17$

Гипотеза $H_1$: $\mu = 17.5$ 

$\alpha = 0.05$

n = 100

Дисперсия $D(X) = 4, \sigma = 2$

Так как у нас известна дисперсия, можем использовать z-критерий.

$$Z_H=\frac{\bar{X}-\mu}{\sigma / \sqrt{n}} = \frac{{17.5}-17}{{2} / \sqrt{100}} $$



In [2]:
import numpy as np

z_h = (17.5 - 17)/(2/np.sqrt(100))
z_h



2.5

In [4]:
import scipy.stats as stats
alpha = 0.05
stats.norm.ppf(1 - alpha)

1.6448536269514722

Так как критерий односторонний, по таблице находим значение z-критерия для $\alpha = 0.05$ и это будет $Z_T = 1.64$. Наблюдаемое значение больше табличного, $Z_H > Z_T, 2.5 > 1.64$, поэтому подтверждается гипотеза $H_1$. 

### Ответ
Утверждение, что шарики имеют средний размер 17мм неверно при $\alpha = 0.05$.

### Здача 5.3
Проведите тест гипотезы. Продавец утверждает, что средний вес пачки печенья составляет 200 г.
Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
202, 203, 199, 197, 195, 201, 200, 204, 194, 190.
Известно, что их веса распределены нормально
Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%? (Провести двусторонний тест.)


Решение:

$H_0: \mu = 200$

$H_1: \mu != 200$

Так как это нормальное распределение, и дисперсия неизвестна, используем критерий Стюдента: 

$$t_H = \dfrac{\overline{X} - a}{\sigma_H / \sqrt{n}}$$

In [18]:
import numpy as np
from scipy import stats

X = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
a = 200
alpha = 0.01
n = len(X)
mean = X.mean()
std = X.std(ddof=1)
(mean, std)

(198.5, 4.453463071962462)

$$t_H = \dfrac{mean - a}{std / \sqrt{n}}$$

In [19]:
t_fact = (mean - a) / std * np.sqrt(n)
t_fact

-1.0651074037450896

In [15]:
t1 = stats.t.ppf(alpha / 2, df=n - 1)
t2 = stats.t.ppf(1 - alpha / 2, df=n - 1)

t1, t2

(-3.2498355440153697, 3.2498355440153697)

По таблице находим значения для критерия Стьюдента. В результате $t_H = -1.06, и -3.2 < t_H < 3.25$,
Значение $t_H$ находится в зоне принятия нулевой гипотезы, поэтому утерждение продавца верно.

### Ответ
Утерждение продавца верно

### Задача 5.4
Рост матерей 172, 177, 158, 170, 178,175, 164, 160, 169, 165

Рост взрослых дочерей: 173, 175, 162, 174, 175, 168, 155, 170, 160

Есть ли статистически значимые различия в росте матерей и дочерей?

### Решение
Принимаем $\alpha = 0.05$
Используем двухвыборочную t-проверку 

$$t=\frac{\overline{x_1}-\overline{x_2}}{\sqrt{\frac{s_1{ }^2}{n_1}+\frac{s_2{ }^2}{n_2}}}$$



In [20]:
x_1 = np.array([172, 177, 158, 170, 178,175, 164, 160, 169, 165])
x_2 = np.array([173, 175, 162, 174, 175, 168, 155, 170, 160])

stats.ttest_ind(x_1, x_2)

Ttest_indResult(statistic=0.24400587215969016, pvalue=0.8101467126543233)

Получаем pvalue > $\alpha$. Вывод: статистически значимых различий в росте матерей и дочерей не обнаружено на уровне значимости $\alpha$ = 0.05